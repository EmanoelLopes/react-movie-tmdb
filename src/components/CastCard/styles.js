import styled from 'styled-components';

export const CastContainer = styled.div`
  background-color: #fff;
  display: flex;
  flex-direction column;
  justify-content: space-between;
  text-align: center;

  h4 {
    color: #fff;
    background-color: rgba(0,0,0,.7);
    padding: 15px;
  }

  img {
    display: block;
    width: 100%;
    height: 100%;
  }
`;

