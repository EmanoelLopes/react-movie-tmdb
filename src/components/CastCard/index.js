import React from 'react';
import { number, string } from 'prop-types';
import { images } from 'utils/styled';
import { CastContainer } from './styles';

const CastCard = props => {
  const {
    id,
    profile,
    name,
    character
  } = props;

  return (
    <CastContainer key={id}>
      <img
        src={(profile)
            ? `${images.path}${images.profile}${profile}`
            : `${images.placeholder.profile}`
        }
        alt={name}
      />
      <h4>{`${name} (${character})`}</h4>
    </CastContainer>
  );
};

CastCard.propTypes = {
  id: number.isRequired,
  profile: string,
  name: string.isRequired,
  character: string.isRequired,
};

CastCard.defaultProps = {
  profile: `${images.placeholder.profile}`,
};

export default CastCard;