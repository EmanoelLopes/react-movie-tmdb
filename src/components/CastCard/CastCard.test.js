import React from 'react';
import { mount } from 'enzyme';
import CastCard from 'components/CastCard';

let wrapped;
let defaultProps = {
  id: 123,
  profile: 'https://placehold.jp/154x250.png?text=NO%20IMAGE',
  name: 'Kit Harington',
  character: 'Jon Snow',
};

beforeEach(() => {
  wrapped = mount(<CastCard {...defaultProps} />);
});

afterEach(() => {
  wrapped.unmount();
});

describe('<CastCard />', () => {
  it('should match snapshot', () => {
    expect(wrapped).toMatchSnapshot();
  });
});
