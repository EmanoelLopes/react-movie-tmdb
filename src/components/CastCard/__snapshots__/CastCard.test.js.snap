// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CastCard /> should match snapshot 1`] = `
.c0 {
  background-color: rgba(0,0,0,.7);
  border: 1px solid #081c24;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  text-align: center;
}

.c0 h4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 15px;
  width: 100%;
}

.c0 img {
  display: block;
  width: 100%;
  height: 100%;
  max-width: 100px;
}

@media (min-width:62em) {
  .c0 {
    max-width: 230px;
  }
}

<CastCard
  character="Jon Snow"
  id={123}
  name="Kit Harington"
  profile="https://placehold.jp/154x250.png?text=NO%20IMAGE"
>
  <styled.div
    key="123"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bxivhb",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  background-color: rgba(0,0,0,.7);
  border: 1px solid ",
              "#081c24",
              ";
  display: flex;
  width: 100%;
  text-align: center;

  ",
              "
    @media (min-width: ",
              "62",
              "em) {
      ",
              "
    max-width: 230px;
  ",
              "
    }
  ",
              "

  h4 {
    align-items: center;
    color: ",
              "#fff",
              ";
    display: flex;
    justify-content: center;
    padding: 15px;
    width: 100%;
  }

  img {
    display: block;
    width: 100%;
    height: 100%;
    max-width: 100px;
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bxivhb",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        <img
          alt="Kit Harington"
          src="https://image.tmdb.org/t/p/w154https://placehold.jp/154x250.png?text=NO%20IMAGE"
        />
        <h4>
          Kit Harington
          <br />
          Jon Snow
        </h4>
        <p />
      </div>
    </StyledComponent>
  </styled.div>
</CastCard>
`;
